### PR TITLE
This allows arguments to be visited and hence modified

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

File: src/test/groovy/graphql/analysis/QueryTraverserTest.groovy [hunk 1]
```diff
@@ -121,6 +127,52 @@ class QueryTraverserTest extends Specification {
 
     }
 
+    def "test postOrder for visitArgs"() {
```
Comment (andimarek) [inline src/test/groovy/graphql/analysis/QueryTraverserTest.groovy]: we should test postOrder and preOrder

File: src/main/java/graphql/analysis/QueryVisitorFieldArgumentInputValueImpl.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package graphql.analysis;
+
+import graphql.Internal;
+import graphql.PublicApi;
+import graphql.language.Value;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputType;
+
+@PublicApi
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFieldArgumentInputValueImpl.java]: The Impl is internal I would say

File: src/main/java/graphql/analysis/QueryVisitorFieldArgumentInputValue.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import graphql.language.Value;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputType;
+
+/**
+ * This describes the tree structure that forms from a argument input type,
+ * especially with `input ComplexType { ....}` types that might in turn contain other complex
+ * types and hence form a tree of values.
+ */
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFieldArgumentInputValue.java]: PublicAPI

File: src/main/java/graphql/analysis/QueryVisitorFieldArgumentEnvironment.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import graphql.language.Argument;
+import graphql.language.Node;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraverserContext;
+
+import java.util.Map;
+
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFieldArgumentEnvironment.java]: PublicAPI

File: src/main/java/graphql/analysis/QueryVisitorFieldArgumentValueEnvironment.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package graphql.analysis;
+
+import graphql.language.Node;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLSchema;
+import graphql.util.TraverserContext;
+
+import java.util.Map;
+
```
Comment (andimarek) [inline src/main/java/graphql/analysis/QueryVisitorFieldArgumentValueEnvironment.java]: PubliAPI?

File: src/test/groovy/graphql/analysis/QueryTraverserTest.groovy (hunk 1 above)
Comment (bbakerman) [reply src/test/groovy/graphql/analysis/QueryTraverserTest.groovy]: done

**Pull request merged**